#! /bin/sh -e

# DP: Allow hwcap's to be disabled with the existence of a file. This
# DP: makes it easier to do upgrades with optimized (hwcap) library
# DP: packages.

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

--- sysdeps/generic/dl-sysdep.c~	2002-09-10 23:22:00.000000000 -0400
+++ sysdeps/generic/dl-sysdep.c	2002-09-10 23:22:17.000000000 -0400
@@ -280,10 +280,14 @@
   struct r_strlenpair *rp;
   char *cp;
 
-  /* Count the number of bits set in the masked value.  */
-  for (n = 0; (~((1UL << n) - 1) & masked) != 0; ++n)
-    if ((masked & (1UL << n)) != 0)
-      ++cnt;
+  if (__access ("/etc/ld.so.nohwcap", F_OK) != 0)
+    /* Disable hwcap's */
+    cnt = 0;
+  else
+    /* Count the number of bits set in the masked value.  */
+    for (n = 0; (~((1UL << n) - 1) & masked) != 0; ++n)
+      if ((masked & (1UL << n)) != 0)
+        ++cnt;
 
   if (cnt == 0)
     {
